tracks/go/exercises/raindrops/README.md

```diff
@@ -0,0 +1,65 @@
+# Raindrops
+
+This is the third core exercise in the go track to receive feedback from a mentor. It is less about learning new concepts but more about deepening the already acquired knowledge. To some the conversion of an `int` to `string` and the modulus operator `%` are new.
+
+This may be an excellent place to talk about the go tooling again -- especially if the tools were not applied (`gofmt`, `golint`, etc).
+
+## Reasonable solutions
+
+The exercise can be solved with `if` statements and `string` concatenation:
+
+
+```
+// Convert implements raindrop speech
+func Convert(i int) string {
+	var res string
+	if i%3 == 0 {
+		res += "Pling"
+	}
+	if i%5 == 0 {
+		res += "Plang"
+	}
+	if i%7 == 0 {
+		res += "Plong"
+	}
+
+	if res == "" {
+		res = strconv.Itoa(i)
+	}
+
+	return res
+}
+```
+
+## Common suggestions
+
+The most common feedback revolves around:
+
+* General programming things (working with strings, map iteration)
+* Readability (formatting, comments, linting)
+
+**Map: random iteration** 
+* are they defining static values (map, slice) in the function? Suggest moving them to package level.
+* are they using a `map` to hold the cases? This gets complicated as they need to somehow make sure the order is correct. From the [specs](https://golang.org/ref/spec#For_statements): "The iteration order over maps is not specified and is not guaranteed to be the same from one iteration to the next." You could direct them towards using `for i := 3; i < 8; i += 2` or a for loop that iterates the slice `[]int{3, 5, 7}`
+
+**Control flows**
+* are they using `if { return ..} else { return ..}`? Point them to `golint` which will suggest to remove the else and outdent its block.
+* are they using a `for i; .. i++` loop? Check what range they are iterating. The most optimal loop iterates `for i := 3; i < 8; i += 2`. Iterating to given `number` (as many do) is not necessary.
+
+**Strings: concatenation and conversion**
+* are they using `strconv.FormatInt`? You can point them to `strconv.Itoa` as being more convenient (no `int64` conversion needed, no second parameter).
```

Comment:
are they using `fmt.Sprintf("%d", int)`. You can mention that while this will convert the integer to a string, it will do no type checking unlike `strconv.Itoa()`

Reply:
Nice! Will add that!

Reply:
although there are type checkings in the newest go version when running tests. Also linters will tell you that you are using the wrong type. So I will focus on the speed and reflection needed form `fmt.Sprintf` as explanation.